Incorrect terra-icon dependency for terra-application-navigation

# Bug Report

## Description
<!-- A clear and concise description of what the bug is. -->
<!-- Providing a link to a live example / minimal demo of the problem greatly helps us debug issues. -->
Terra Application Navigation has a Terra Icon dependency of 3.1.0 (see [here](https://github.com/cerner/terra-framework/blob/master/packages/terra-application-navigation/package.json#L47)) but it uses [IconExtensions](https://github.com/cerner/terra-core/blob/master/packages/terra-icon/src/icon/IconExtensions.jsx) that was introduced in terra-icon version 3.11.0. I believe this may have been a typo in the dependency version.

Due to my project requiring terra-icon <= 3.10.0 this throws an error when attempting to build:

`Module not found: Error: Can't resolve 'terra-icon/lib/icon/IconExtensions' in '....../node_modules/terra-application-navigation/lib/extensions'`


## Additional Context / Screenshots
<!-- Add any other context about the problem here. If applicable, add screenshots to help explain. -->

## Expected Behavior
<!-- A clear and concise description of what you expected to happen. -->

## Possible Solution
<!--- If you have suggestions to fix the bug, let us know -->
Update [terra-application-navigation package.json](https://github.com/cerner/terra-framework/blob/master/packages/terra-application-navigation/package.json#L47) terra-icon dependency to 3.11.0.

## Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Component Name and Version: terra-application-navigation
* Browser Name and Version: Chrome 77.0.3865.90
* Node/npm Version: [e.g. Node 8/npm 5] npm 6.1.0
* Webpack Version: 4.41.0
* Operating System and version (desktop or mobile): macOS Mojave 10.14.6

## @ Mentions
<!-- @ Mention anyone on the terra team that you have been working with so far. -->

--- a/packages/terra-application-navigation/CHANGELOG.md
+++ b/packages/terra-application-navigation/CHANGELOG.md
@@ -4,6 +4,7 @@ Changelog
 Unreleased
 ----------
 ### Changed
+* Update terra-icon package version.
 * Adjusted title margin and padding
 * Update tests and screenshots to support theme tests
 

--- a/packages/terra-application-navigation/package.json
+++ b/packages/terra-application-navigation/package.json
@@ -44,7 +44,7 @@
     "terra-button": "^3.3.0",
     "terra-content-container": "^3.0.0",
     "terra-doc-template": "^2.2.0",
-    "terra-icon": "^3.1.0",
+    "terra-icon": "^3.11.0",
     "terra-image": "^3.0.0",
     "terra-overlay": "^3.0.0",
     "terra-popup": "^6.13.0",

